### DISCUSSION TITLE
Vue integration

### DISCUSSION THREAD

**Discussion by johnwason** [pyodide/pyodide#3117] [Show and tell]
I am using Vue for a project, and created a wrapper for Vue objects that neatly integrates into Python objects.

https://github.com/johnwason/pyri-webui-browser/blob/vue_module/src/pyri/webui_browser/vue.py

Example components:

https://github.com/johnwason/pyri-webui-browser/tree/vue_module/src/pyri/webui_browser/components

I haven't written any documentation yet on how to use the wrapper. Essentially, attributes and decorators are used to build the Vue objects.

I can break this up into a separate module, but I'd rather hand it off to someone else if they want to maintain it.

**Comment by johnwason** [comment]
We are also using golden layout, and this is a Vue implementation that works with components:

https://github.com/pyri-project/pyri-webui-browser/blob/master/src/pyri/webui_browser/golden_layout.py

**Comment by ryanking13** [comment]
Thanks for sharing your work @johnwason! It is nice to see Pyodide being integrated into other frameworks.
